protostar/commands/invoke_command.py

```diff
@@ -9,28 +10,51 @@
     ProtostarArgument,
     ProtostarCommand,
     SignableCommandUtil,
+    MessengerFactory,
 )
+from protostar.cli.common_arguments import BLOCK_EXPLORER_ARG
+from protostar.io import StructuredMessage, LogColorProvider
 from protostar.protostar_exception import ProtostarException
 from protostar.starknet_gateway import (
     GatewayFacadeFactory,
     SuccessfulInvokeResponse,
     create_block_explorer,
-    SupportedBlockExplorerName,
-    NetworkConfig,
-    BlockExplorer,
 )
 from protostar.starknet_gateway.gateway_facade import Fee
-from protostar.cli.common_arguments import BLOCK_EXPLORER_ARG
+
+
+@dataclass
+class SuccessfulInvokeMessage(StructuredMessage):
+    response: SuccessfulInvokeResponse
+    tx_url: Optional[str]
+
+    def format_human(self, fmt: LogColorProvider) -> str:
+        message = f"""\
+Invoke transaction was sent.
+Transaction hash: 0x{self.response.transaction_hash:064x}
+"""
+        if self.tx_url:
+            message += f"{self.tx_url}\n"
+
+        return message
+
+    def format_dict(self) -> dict:
+        return {
```

Comment:
why no response in the dict?

Reply:
because I am formatting tx hash as hex string

Reply:
ah ok i thought it contained something more. my bad